v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstogcs/BigtableChangeStreamsToGcsIT.java

```diff
@@ -0,0 +1,550 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.bigtablechangestreamstogcs;
+
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatPipeline;
+
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.cloud.teleport.bigtable.ChangelogEntry;
+import com.google.cloud.teleport.bigtable.ChangelogEntryJson;
+import com.google.cloud.teleport.bigtable.ModType;
+import com.google.cloud.teleport.metadata.SkipDirectRunnerTest;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import org.apache.beam.it.common.PipelineLauncher.JobState;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineLauncher.LaunchInfo;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.common.PipelineOperator.Config;
+import org.apache.beam.it.common.TestProperties;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.artifacts.Artifact;
+import org.apache.beam.it.gcp.bigtable.BigtableResourceManager;
+import org.apache.beam.it.gcp.bigtable.BigtableResourceManagerCluster;
+import org.apache.beam.it.gcp.bigtable.BigtableTableSpec;
+import org.apache.beam.it.gcp.storage.GcsResourceManager;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+/** Integration test for {@link BigtableChangeStreamsToGcs}. */
+@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})
+@TemplateIntegrationTest(BigtableChangeStreamsToGcs.class)
+@RunWith(JUnit4.class)
+public final class BigtableChangeStreamsToGcsIT extends TemplateTestBase {
+
+  public static final String SOURCE_COLUMN_FAMILY = "cf";
+  private static final Duration EXPECTED_REPLICATION_MAX_WAIT_TIME = Duration.ofMinutes(10);
+  private static final String TEST_ZONE = "us-central1-a";
+  private static BigtableResourceManager bigtableResourceManager;
+  private static GcsResourceManager gcsResourceManager;
```

Comment:
These being static may cause issues. We'll see if gitactions pass

Reply:
+1, this will surely cause issues. There are many tests in the same class, and we parallelize in GH Actions